### PR TITLE
Add additional null safety checks and exception handling to PipelineWitness

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs [hunk 1]
```diff
@@ -63,7 +63,37 @@ public async Task ProcessRunAsync(Uri runUri)
...
+
+            if (build == null)
+            {
+                this.logger.LogWarning("Unable to process run due to missing build");
+                return;
+            }
+
+            if (build.Deleted)
+            {
+                this.logger.LogWarning("Skipping deleted build");
```
Comment (benbp) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs:86]: Not sure if this is a valid property, but in the case that the log context doesn't already add this to the output:

```suggestion
                this.logger.LogWarning($"Skipping deleted build {build.Id}");
```

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs [hunk 2]
```diff
@@ -63,7 +63,37 @@ public async Task ProcessRunAsync(Uri runUri)
...
+            {
+                build = await buildClient.GetBuildAsync(projectGuid, runId);
+            }
+            catch (BuildNotFoundException)
+            {
+            }
+
+            if (build == null)
+            {
+                this.logger.LogWarning("Unable to process run due to missing build");
```
Comment (benbp) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs:80]: In the case that this isn't already in the log context.

```suggestion
                this.logger.LogWarning($"Unable to process run due to missing build for run {runId}");
```

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs [hunk 3]
```diff
@@ -63,7 +63,37 @@ public async Task ProcessRunAsync(Uri runUri)
...
+            if (build.Deleted)
+            {
+                this.logger.LogWarning("Skipping deleted build");
+                return;
+            }
+
+
+            if (build.StartTime == null || build.FinishTime == null)
+            {
+                this.logger.LogWarning("Skipping build with null start or finish time");
```
Comment (benbp) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs:93]:
```suggestion
                this.logger.LogWarning($"Skipping build {build.Id} with null start or finish time");
```

Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs:93]: I'll use structured logging here:

```cs
{
    this.logger.LogWarning("Skipping build with null start or finish time. Project: {Project}, BuildId: {BuildId}", project.Name, runId);
}

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs (hunk 1 above)
Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/RunProcessor.cs]: added via structured log message

**Review by benbp — APPROVED** [review]

**Pull request merged**